cli/src/ui.rs

```diff
@@ -632,14 +621,53 @@ impl Ui {
             Some(false) => "(yN)",
             None => "(yn)",
         };
-        let default_index = default.map(|c| if c { 0 } else { 1 });
-
-        let index = self.prompt_choice(
+        self.prompt_choice_with(
             &format!("{prompt} {default_str}"),
-            &["y", "n", "yes", "no", "Yes", "No", "YES", "NO"],
-            default_index,
-        )?;
-        Ok(index % 2 == 0)
+            default.map(|v| (v, if v { "y" } else { "n" })),
+            |input| {
+                if input.eq_ignore_ascii_case("y") || input.eq_ignore_ascii_case("yes") {
+                    Ok(true)
+                } else if input.eq_ignore_ascii_case("n") || input.eq_ignore_ascii_case("no") {
+                    Ok(false)
+                } else {
+                    Err("unrecognized response")
+                }
+            },
+        )
+    }
+
+    /// Repeats the given prompt until `parse(input)` returns a value.
+    ///
+    /// If the default `(value, text)` is specified, an empty input will be
+    /// mapped to the default `value`. The default `value` will also be used in
+    /// non-interactive session. The `text` will be printed in that case.
+    pub fn prompt_choice_with<T, E: fmt::Display>(
+        &self,
+        prompt: &str,
+        default: Option<(T, &str)>,
```

Comment:
Perhaps it'd be simpler to understand this function if `default` was `Option<&str>`, and the default path evaluated `parse(default.unwrap()`? It would make it very slightly less flexible, but I think also less confusing. It'd also match what would have happened in the interactive case if the user typed in the default value.

Reply:
Good point, done.